<!-- -->
<launch>

	<arg name="pointcloud" default="point_map" /> <!--/rtabmap/cloud_map   point_map -->
	<arg name="prefix" default="indires_rover" />
	<arg name="robot_topic" default="/$(arg prefix)/odom_filtered" />
	<arg name="sensor_frame" default="$(arg prefix)/front_rgbd_camera/depth_optical_frame" />

	<!-- launch-prefix="xterm -e gdb (doble hyphen)args"-->
	 <node pkg="pcl_filters" type="pcl_filters_node" name="pcl_filters_node_1" output="screen" >

		<param name="pointcloud_topic" value="$(arg pointcloud)" /> <!-- "indires_rover/front_rgbd_camera/front_rgbd_camera/depth/points" -->
		<param name="output_topic" value="rrt_pointcloud_samples" />
		<param name="use_robot_pose" value="true" />
		<param name="robot_pose_topic" value="$(arg robot_topic)" />
		<param name="odom_frame" value="$(arg prefix)/odom" /> <!--the output pointcloud is published in this frame-->
		<param name="sensor_frame" value="$(arg sensor_frame)" />
		<param name="local_grid_radius" value="5.0" /> <!--5.0  for nfe,bfe: 12.0-->

		<param name="filter_ceiling" value="true" />
		<param name="show_normal_arrows" value="false" />
		<param name="pitch_max_inclination" type="double" value="0.80" /> <!--0.87 = 50ยบ  0.79 rad = 45ยบ -->
		<param name="roll_max_inclination" type="double" value="0.80" />
		<param name="max_roughness" type="double" value="0.8" /> <!--0.03-->
		<param name="max_height_allowed" type="double" value="1.4" />

		<!--remap from="/upo_navigation_simple/goal" to="/move_base_simple/goal" /-->
		<rosparam file="$(find pcl_filters)/launch/rrt_filter_params.yaml" command="load"  />

	</node>

	<!-- launch-prefix="xterm -e gdb (doble hyphen)args"-->
	 <node pkg="pcl_filters" type="pcl_filters_node" name="pcl_filters_node_2" output="screen" >

		<param name="pointcloud_topic" value="$(arg pointcloud)" /> <!-- "indires_rover/front_rgbd_camera/front_rgbd_camera/depth/points" -->
		<param name="output_topic" value="features_pointcloud" />
		<param name="use_robot_pose" value="true" />
		<param name="robot_pose_topic" value="$(arg robot_topic)" />
		<param name="odom_frame" value="$(arg prefix)/odom" /> <!--"indires_rover/odom"-->
		<param name="local_grid_radius" value="5.0" /> <!--5.0  for nfe,bfe: 12.0-->

		<param name="filter_ceiling" value="false" />
		<param name="show_normal_arrows" value="false" />

		<!--remap from="/upo_navigation_simple/goal" to="/move_base_simple/goal" /-->
		<rosparam file="$(find pcl_filters)/launch/3dfeatures_params.yaml" command="load"  />

	</node>


	 <node pkg="pcl_filters" type="pcl_filters_node" name="pcl_filters_node_3" output="screen" >

		<param name="pointcloud_topic" value="$(arg pointcloud)" /> <!-- "indires_rover/front_rgbd_camera/front_rgbd_camera/depth/points" -->
		<param name="output_topic" value="local_control_pointcloud" />
		<param name="use_robot_pose" value="true" />
		<param name="robot_pose_topic" value="$(arg robot_topic)" />
		<param name="odom_frame" value="$(arg prefix)/odom" /> <!--"indires_rover/odom"-->
		<param name="local_grid_radius" value="1.6" />

		<param name="filter_ceiling" value="false" />
		<param name="show_normal_arrows" value="false" />

		<!--remap from="/upo_navigation_simple/goal" to="/move_base_simple/goal" /-->
		<rosparam file="$(find pcl_filters)/launch/local_control_params.yaml" command="load"  />

	</node>


	<node pkg="pcl_filters" type="pcl_filters_node" name="pcl_filters_node_4" output="screen" >

		<param name="pointcloud_topic" value="$(arg pointcloud)" /> <!-- "indires_rover/front_rgbd_camera/front_rgbd_camera/depth/points" -->
		<param name="use_service" value="true" />
		<param name="publish_topic" value="false" />
		<param name="use_robot_pose" value="true" />
		<param name="robot_pose_topic" value="$(arg robot_topic)" />
		<param name="odom_frame" value="$(arg prefix)/odom" /> <!--"indires_rover/odom"-->
		<param name="local_grid_radius" value="7.0" /> <!--7.0  for nfe,bfe: 14.0-->

		<param name="filter_ceiling" value="false" />
		<param name="show_normal_arrows" value="false" />

		<!--remap from="/upo_navigation_simple/goal" to="/move_base_simple/goal" /-->
		<rosparam file="$(find pcl_filters)/launch/exp_params.yaml" command="load"  />

	</node>



</launch>

